### ISSUE TITLE
[YSQL] Use serializable isolation for simple UPDATE statement

### ISSUE THREAD

**Issue by robertpang** [yugabyte/yugabyte-db#1444]
Currently when updating a table like this:

```
postgres=# create table t (k int primary key, v1 int, v2 int);
postgres=# insert into t values (1, 2, 3);
postgres=# select * from t;
 k | v1 | v2
---+----+----
 1 |  2 |  3
(1 row)
postgres=# update t set v1 = null where k = 1;
```
We issue the following write request that reads the existing columns, compares the values to be written and writes only those columns that are modified.
```
YQL_CLIENT_PGSQL
stmt_type: PGSQL_UPDATE
table_id: "000030a0000030008000000000004000"
schema_version: 0
hash_code: 4624
ybctid_column_value {
  value {
    binary_value: "G\022\020H\200\000\000\001!!"
  }
}
column_new_values {
  column_id: 0
  expr {
    value {
      int32_value: 1
    }
  }
}
column_new_values {
  column_id: 1
  expr {
    value {
    }
  }
}
column_new_values {
  column_id: 2
  expr {
    value {
      int32_value: 3
    }
  }
}
column_refs {
  ids: 0
  ids: 1
  ids: 2
}
```
Apart from incurring a small read overhead, it requires snapshot isolation and incurs unnecessary read/write conflicts (say with another UPDATE statement like `update t set v2 = null where k = 1`. We should issue such write as pure-write operation with serializable isolation to allow the 2 UPDATE statements to execute concurrently.

**Comment by ndeodhar** [comment]
@m-iancu This is also something that came up during TPCC analysis. Will be good to commit this diff.

**Comment by m-iancu** [comment]
Fixed by 7e222b4.
